packages/components/src/utils/useContainerQueries.tsx

```diff
@@ -0,0 +1,114 @@
+import type React from 'react'
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import { useDebouncedCallback } from 'use-debounce'
+
+const DEFAULT_DEBOUNCE_MS = 1000
+
+/**
+ * Tailwind CSS default container query breakpoints in pixels
+ * These match the default values from @tailwindcss/container-queries plugin
+ */
+const DEFAULT_BREAKPOINTS = {
+  'xs': 320,
+  'sm': 384,
+  'md': 448,
+  'lg': 512,
+  'xl': 576,
+  '2xl': 672,
+  '3xl': 768,
+  '4xl': 896,
+  '5xl': 1024,
+  '6xl': 1152,
+  '7xl': 1280,
+} as const
+
+type ContainerQueries = {
+  isXsOrLarger: boolean
+  isSmOrLarger: boolean
+  isMdOrLarger: boolean
+  isLgOrLarger: boolean
+  isXlOrLarger: boolean
+  is2xlOrLarger: boolean
+  is3xlOrLarger: boolean
+  is4xlOrLarger: boolean
+  is5xlOrLarger: boolean
+  is6xlOrLarger: boolean
+  is7xlOrLarger: boolean
+}
+
+export const useContainerQueries = (): {
```

Comment:
This doesn't have support for custom queries or components compared to useMediaQueries. I think the default is already pretty good and we can always add support when needed. For `components` support, I think not really needed as they could just use tailwind directly

<img width="507" height="365" alt="Image" src="https://github.com/user-attachments/assets/f37a0c3c-ce41-456b-b3f9-30c2410f1e48" />